workflow/workflow.go

```diff
@@ -6,32 +6,44 @@ import (
 	validator "gopkg.in/go-playground/validator.v9"
 )
 
+// ID is the ID of the Result's node
+func (r Result) ID() string {
+	return "trigger:result"
```

Comment:
we could let the user set its own key on the triggers but as it is unique for now it's not necessary. But happy to change if you have any problem with this hardcoded value

Reply:
I definitely suggest to force the node to define the key. Otherwise the client (cli) have also to know that this specific type of node has this specific key during the compilation of the workflow.

Reply:
done